### ISSUE TITLE
[Snowflake] CALL statement with a colon-prefixed bind variable inside BEGIN...END causes PRS error

### ISSUE THREAD

**Issue by miguelmmu** [sqlfluff/sqlfluff#8242]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting/parsing a Snowflake Scripting block (DECLARE ... BEGIN ... END, either standalone or inside CREATE TASK/CREATE PROCEDURE) that contains a CALL statement whose argument is a bind variable prefixed with a colon (e.g. CALL my_proc(:my_var)), SQLFluff fails to parse the statement. This is the officially documented way to pass a Snowflake Scripting variable as an argument to CALL (see [Snowflake docs](https://docs.snowflake.com/en/developer-guide/stored-procedure/stored-procedures-snowflake-scripting#calling-a-stored-procedure-without-using-the-returned-value)): "If you need to pass in any variables or arguments as input arguments in the CALL statement, remember to use a colon (:) in front of the variable name."

The same bind variable syntax (:var) works fine in other statements inside the same block (e.g. SELECT :var;), and CALL works fine if the colon is removed — so the issue is specific to the combination of CallStatementSegment + a colon-prefixed variable reference as an argument.

### Expected Behaviour

SQLFluff should parse a CALL statement whose arguments include colon-prefixed bind/scripting variables (:var) when used inside a Snowflake Scripting BEGIN ... END block, since this is valid and officially documented Snowflake syntax. No PRS error should be raised.

### Observed Behaviour

Linting the query below raises a PRS error:

```sql
DECLARE
  load_timestamp VARCHAR DEFAULT CURRENT_TIMESTAMP();
BEGIN
  CALL my_schema.my_procedure(:load_timestamp);
END;
``` 
```
violations:
  - start_line_no: 4
    start_line_pos: 3
    code: PRS
    description: "Line 4, Position 3: Found unparsable section: 'CALL my_schema.my_procedure(:load_timest...'"
    name: ''
    warning: false
``` 

For comparison, these related cases all parse correctly, which narrows the bug down specifically to CALL + :variable:
```sql
-- OK: same variable used with SELECT instead of CALL
BEGIN
  SELECT :load_timestamp;
END;

-- OK: CALL with the variable but without the colon prefix (non-valid syntax, shown only for isolation)
BEGIN
  CALL my_schema.my_procedure(load_timestamp);
END;
``` 

### How to reproduce

Run sqlfluff lint (or sqlfluff parse) on the SQL block above with --dialect snowflake.

### Dialect

snowflake

### Version

4.2.2

### Configuration

dialect = snowflake
templater = raw
encoding = utf-8
max_line_length = 0
large_file_skip_byte_limit = 0

rules = AL02, AL03, AL04, AL05, AL08, AM01, AM02, AM04, AM05, AM06, AM07, CV05, CV07, CV08, CV12, LT07, LT10, LT11, LT13, LT14, RF01, RF02, RF04, RF05, RF06, ST02, ST03, ST04, ST07, ST08, ST10, ST11, ST12

indent_unit = space
tab_space_size = 4
indented_joins = False
indented_ctes = False
indented_using_on = True
indented_on_contents = False
indented_then = False
indented_then_contents = False
implicit_indents = allow
template_blocks_indent = True

[sqlfluff:rules:references.qualification]
subqueries_ignore_external_references = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
There is a recently merged (not yet released) PR on this: #8163

I have just checked, and the main branch is properly parsing now the above example.
So the next release will solve this issue.

**Comment by miguelmmu** [comment]
Sorry I missed that PR.
